docs/architecture.md

```diff
@@ -30,12 +30,21 @@ def mount(main: click.Group, **kwargs: Any) -> None:
     main.add_command(my_command)
 ```
 
-### Contexts
+### Pulp Glue
+
+Pulp CLI uses `pulp-glue` library to manage plugin versions dependent features. Historically, those restrictions were
```

Comment:
This is not the reason, we introduced glue. Glue is an abstraction layer that let's you perform high level operations in pulp (like repository.update(...)). It abstracts talking to the rest api including parsing the api.json, waiting on tasks and task groups. And it allows multiple projects to use this level of abstraction.
And from that architectural viewpoint it's natural to put all the version dependent fixes there too.

Historically there was no glue and therefore not a good separation of concerns. But i doubt, that this information is really helpful.

"Thus, `pulp-glue` aims to be the comprehensive guide to all known Pulp API subtleties." This is not the selling point for pulp-glue, but rather a corollary.

Reply:
> Pulp CLI uses `pulp-glue` library as an abstraction layer that let's you perform high level operations in pulp.
> It abstracts talking to the rest api (including parsing the api.json), waiting on tasks and task groups.
> Also, it allows multiple projects to use this level of abstraction, such as `pulp-smash` and `pulpcore`.
> For this, `pulp-glue` is becoming the comprehensive guide to all known Pulp API subtleties, such as version dependent fixes.
>